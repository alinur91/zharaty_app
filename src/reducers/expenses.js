const expensesReducerDefaultState=[]

const expensesReducer = (state=expensesReducerDefaultState,action)=>{
 switch (action.type) {
  case 'ADD_EXPENSE':
    return [...state,action.expense]
   break;

     case 'REMOVE_EXPENSE':
      return state.filter(({id})=> id !== action.id)
   break;

     case 'EDIT_EXPENSE':
       return state.map((expense)=> expense.id === action.id?{...expense,...action.updates}:expense)
      //  const needsToEdit = state.find(({id})=>id===action.id)
      //  return [...state,{...needsToEdit,...action.updates}]
   
   break;

  default:
   return state
   break;
 }
}

export default expensesReducer