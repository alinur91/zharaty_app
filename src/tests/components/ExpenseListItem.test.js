import React, { Component } from 'react'
import {shallow} from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses'


test('render single expense',()=>{
 const wrapper = shallow(<ExpenseListItem {...expenses[1]}/>)
 expect(wrapper).toMatchSnapshot()
})