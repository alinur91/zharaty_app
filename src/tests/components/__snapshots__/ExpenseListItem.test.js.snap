// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render single expense 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h3>
      nitrogen
    </h3>
  </Link>
  <p>
    $1,099.00
    -
    December 28th, 1969
  </p>
</div>
`;
